src/components/App/ImportStudyList.js

```diff
@@ -0,0 +1,168 @@
+            if (!this.state.studyListText.match(/\d{5}/g)) {
+                openSnackbar('error', 'Cannot import an empty/invalid Study List.');
+                return;
+            }
+            const currSchedule = AppStore.getCurrentScheduleIndex();
+            const sectionCodes = this.state.studyListText
+                .split('\n')
+                .map((line) => line.match(/\d{5}/g))
+                .filter((id) => id)
+                .flat();
+            let sectionsAdded = 0;
+            try {
+                (
+                    await Promise.all(
+                        sectionCodes
+                            .reduce((result, item, index) => {
+                                // WebSOC queries can have a maximum of 8 course codes in tandem
+                                const chunkIndex = Math.floor(index / 8);
+                                result[chunkIndex] ? result[chunkIndex].push(item) : (result[chunkIndex] = [item]);
+                                return result;
+                            }, []) // https://stackoverflow.com/a/37826698
+                            .map((sectionCode) =>
+                                queryWebsoc({ term: this.state.selectedTerm, sectionCodes: sectionCode.join(',') })
+                            )
+                    )
+                ).forEach((response) => {
+                    response.schools
+                        .map((school) => school.departments)
+                        .flat()
+                        .map((dept) => dept.courses)
+                        .flat()
+                        .forEach((course) => {
+                            course.sections.forEach((section) => {
+                                addCourse(section, course, this.state.selectedTerm, currSchedule);
+                                ++sectionsAdded;
+                            });
+                        });
+                });
+                if (sectionsAdded === sectionCodes.length) {
+                    openSnackbar('success', `Successfully imported ${sectionsAdded} of ${sectionsAdded} classes!`);
+                } else if (sectionsAdded !== 0) {
+                    openSnackbar(
+                        'warning',
+                        `Successfully imported ${sectionsAdded} of ${sectionCodes.length} classes. 
+                        Please make sure that you selected the correct term and that none of your classes are missing.`
+                    );
+                } else {
+                    openSnackbar(
+                        'error',
+                        'Failed to import any classes! Please make sure that you pasted the correct Study List.'
+                    );
+                }
+            } catch (e) {
+                this.handleError(e);
+            }
+        }
+    };
+
+    enterEvent = (event) => {
+        const charCode = event.which ? event.which : event.keyCode;
+        if (charCode === 13 || charCode === 10) {
```

Comment:
Can you add a comment for what these charCodes are? I figured out that 13 is enter but in the first two references I looked at I couldn't find what charCode 10 is and had to use String.fromCharCode to figure out it was newline.